src/components/TradeWidget/TokenRow.tsx

```diff
@@ -131,6 +141,36 @@ const TokenRow: React.FC<Props> = ({
     setValue(inputId, formatAmountFull(balance.exchangeBalance, balance.decimals, false))
   }
 
+  const enforcePrecision = useCallback(() => {
+    const newValue = adjustPrecision(inputValue, selectedToken.decimals)
+    if (inputValue !== newValue) {
+      setValue(inputId, newValue, true)
+    }
+  }, [inputValue, selectedToken.decimals, setValue, inputId])
+
+  useEffect(() => {
+    enforcePrecision()
+  }, [enforcePrecision])
+
+  const removeExcessZeros = useMemo(
+    () => (event: React.SyntheticEvent<HTMLInputElement>): void => {
+      // Q: Why do we need this function instead of relying on `preventInvalidChars` or `enforcePrecision`?
+      // A: Because on those functions we still want the user to be able to input partial values. E.g.:
+      //    0 -> 0. -> 0.1 -> 0.10 -> 0.105
+      //    When losing focus though, we remove everything that's redundant, such as leading zeros, trailing dots and/or zeros
+      // Q: Why not use formatAmount/parseAmount that already take care of this?
+      // A: Too many steps (convert to and from BN) and binds the function to selectedToken.decimals
+
+      const { value } = event.currentTarget
+      const newValue = value.replace(leadingAndTrailingZeros, '').replace(trailingZerosAfterDot, '$1')
```

Comment:
One replacement with `value.replace(/^0{2,}(?=\d*\.?)|\.0*$|(?<=\.\d*)0*$/g, '')`
Check [here](https://regex101.com/r/MismO5/2)

Reply:
Yes, I tried that, but the positive look behind with wildcards doesn't work.
From https://regex101.com/: `* A quantifier inside a lookbehind makes it non-fixed width`

Even then one you linked has an error message: `? The preceding token is not quantifiable`

Reply:
Worked for me in chrome, but actually doesn't in Firefox now that I tried. Don't bother then